File: 0264-ugly-number-ii/0264-ugly-number-ii.cpp
class Solution {
public:
    int nthUglyNumber(int n) {
        set<long> uglyNumSet;
        uglyNumSet.insert(1);

        long currUgly = 1;
        for (int i = 0; i < n; i++) {
            currUgly = *uglyNumSet.begin();
            uglyNumSet.erase(uglyNumSet.begin());

            uglyNumSet.insert(currUgly * 2);
            uglyNumSet.insert(currUgly * 3);
            uglyNumSet.insert(currUgly * 5);
        }

        return static_cast<int>(currUgly);
        
    }
};

File: 0264-ugly-number-ii/0264-ugly-number-ii.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        ugly_num_set = set()
        ugly_num_set.add(1)
        curr_ugly = 1
        for i in range(n):
            curr_ugly = min(ugly_num_set)
            ugly_num_set.remove(curr_ugly)

            ugly_num_set.add(curr_ugly * 2)
            ugly_num_set.add(curr_ugly * 3)
            ugly_num_set.add(curr_ugly * 5)
        
        return curr_ugly